examples: add example of border router with XBee on native platform.

Yonezawa-T2 opened the pull request:
I'm struggling to make my Mac with XBee connected by USB to be a border router of 6LoWPAN. My objective is not fully achieved, but this example is useful to test modification of #4443, #4444, #4445, #4446, and #4447. This example depends on those PRs. The branch https://github.com/Yonezawa-T2/RIOT/tree/xbee_border_router is the master branch merged with those pull requests.

This PR may or may not be broken by #3622. I will investigate later.

authmillenon commented:
I'm sorry I neither knew up until now, that native has a UART interface, nor that much about Xbee. I reassign.

authmillenon commented on examples/gnrc_xbee_border_router/README.md:
```diff
@@ -0,0 +1,117 @@
+in the Makefile. If the baudrate is other than 9600, configure it in
+`xbee_conf.h`.
+
+See `img/page_001.svg` for the organization of modules and devices.
```
You can link images with the `![Caption](link)` syntax. With this notation, people who look at this page in the parsed version wouldn't likely know what you mean ;-)

Yonezawa-T2 replied:
It seems that we cannot preview it on GitHub. Converting image to PNG may help.

kYc0o commented:
Any news on this PR?

kYc0o commented:
AFAIK, all the dependencies on this PR have been merged. Can you rebase and re-test if everything is ok? Thanks for all your contributions!

kYc0o commented on examples/gnrc_xbee_border_router/xbee_params.h:
```diff
@@ -0,0 +1,39 @@
+static xbee_params_t xbee_params[] = {
+    { .uart = XBEE_UART,
+      .baudrate = 9600U,
+      // .baudrate = 115200U,
```
Can you use C style comments? :)

kYc0o replied:
Or even remove the line since it doesn't hurt...

Yonezawa-T2 replied:
Removed.

Yonezawa-T2 commented:
Rebased. You may need #5364 for Clang.

authmillenon commented:
Was already fixed in #5360.

kYc0o commented:
@Yonezawa-T2 sorry for the loooong delay, but I think we're ready to merge this. Can you please rebase to update your PR?

Yonezawa-T2 commented:
Rebased and squashed. I'm worrying about the scenario 1 in README.md. It may no longer work because RIOT respects RFC strictly now. I don't have enough time to test it.

kYc0o commented:
All the dependencies of this PR are merged now. Can you make the needed tests and try to merge it for the next release? It seems a very interesting PR.

Yonezawa-T2 commented:
Updated documentation and fixed minor compilation error. Confirmed all scenarios on README.md works.

miri64 commented:
Postponed due to feature freeze

kYc0o commented:
I was aiming to merge this since longtime ago, IMHO it's a wonderful PR which can help to configure a BR very quickly on OS X, since for now we don't have a easy-to-setup BR solution on this OS.
@Yonezawa-T2 can you rebase to master? and maybe rebase to #6235 (or we can just wait to get it merged).

I successfully configured OS X with the XBee with this PR.

Yonezawa-T2 commented:
@kYc0o, I have rebased and updated it to work with #6235. Not tested with actual XBee devices (I cannot access to them for now).

kYc0o commented:
Well now it hangs while setting a new address... before the rebase and the last commit it worked correctly...

Yonezawa-T2 commented:
- `EXC_BAD_ACCESS` because `vector[0].iov_len` in `xbee_send` is 4294967212 (i.e. (size_t)(-84)) 
- because `xbee_build_hdr` called from `xbee_adpt_send` returns `-EOVERFLOW` (-84)
- because `payload_len` is 104 while `XBEE_MAX_PAYLOAD_LENGTH` is 100 (or 95 with encryption)

Note `xbee_get` returns `XBEE_MAX_PAYLOAD_LENGTH` for `NETOPT_MAX_PACKET_SIZE` correctly.

I don't have enough time to dig deeper. Sorry.

kYc0o commented:
Won't make it for this release unfortunately...

aabadie commented:
What's the status here @Yonezawa-T2 ?

kYc0o commented:
Ping @Yonezawa-T2 !

smlng commented:
IMHO we already  have a good deal of `gnrc_*` examples in our main repo. How about moving this PR to the applications repository? And close here as _memo_?

kYc0o commented:
I'm super interested on re-working this PR. @smlng you might close it as memo if you want. Regarding the examples issue, I'd say we can re-discuss it again when I have the time to re-open this.

@Yonezawa-T2 is not responsive anymore, but anyways thanks for the already (hard) work done!

smlng commented:
closing as memo as suggested, removing milestone as well
